src/Admin/Preview.php

```diff
@@ -21,9 +21,114 @@ function() {
 					$this->registerPreviewStatusFieldsAndMutations();
 				}
 			);
+
+			$use_cloud_loader = self::get_setting( 'use_gatsby_content_sync' );
+			
+			if ( $use_cloud_loader === 'on' ) {
+				add_filter( 'preview_post_link', function( $link, $post ) {
+					return self::get_gatsby_content_sync_url_for_post( $post );
+				}, 10, 2 );
+			}
 		}
 	}
 
+	public static function get_preview_manifest_id_for_post( $post ) {
+		$graphql_single_name = 
+			get_post_type_object( $post->post_type )
+				->graphql_single_name;
+
+		if ( !$graphql_single_name || $graphql_single_name === "" ) {
+			// if we don't have a graphql single name
+			// Gatsby can't use this post anyway.
+			// No need to return a manifest
+			return null;
+		}
+
+		$action_monitor_posts = new \WP_Query( [
+			'post_type'      => 'action_monitor',
+			'post_status'    => 'any',
+			'posts_per_page' => 1,
+			'no_found_rows'  => true,
+			'fields'         => 'ids',
+			'tax_query'      => [
+				'relation' => 'AND',
+				[
+					'taxonomy' => 'gatsby_action_ref_node_dbid',
+					'field'    => 'name',
+					'terms'    => sanitize_text_field( $post->ID ),
+				],
+				[
+					'taxonomy' => 'gatsby_action_ref_node_type',
+					'field'    => 'name',
+					'terms'    => sanitize_text_field( $graphql_single_name ),
+				],
+				[
+					'taxonomy' => 'gatsby_action_stream_type',
+					'field'    => 'name',
+					'terms'    => 'PREVIEW',
+				]
+			],
+		] );
+
+		if (
+			isset( $action_monitor_posts->posts )
+			&& ! empty( $action_monitor_posts->posts )
+		) {
+			$action_monitor_post_id = $action_monitor_posts->posts[0];
+			$referenced_node_preview_data = get_post_meta(
+				$action_monitor_post_id,
+				'_gatsby_preview_data',
+				true
+			);
+
+			$preview_data = false;
+
+			if (
+				$referenced_node_preview_data
+				&& $referenced_node_preview_data !== ""
+			) {
+				$preview_data = json_decode( $referenced_node_preview_data );
+			}
+
+			if (
+				$preview_data
+				&& property_exists( $preview_data, 'manifestIds' )
+				&& count( $preview_data->manifestIds ) > 0
+			) {
+				return $preview_data->manifestIds[0];
+			}
+		} 
+
+		// if the above doesn't return a value we generate a new manifest ID from the post_modified date and post db id
+		$revision = self::getPreviewablePostObjectByPostId( $post->ID );
+		$revision_modified = $revision->post_modified ?? null;
+
+		if ( !$revision_modified || $revision_modified === "" ) {
+			return null;
+		}
+
+		$manifest_id = $post->ID . $revision_modified;
+
+		return $manifest_id;
+	}
+
+	public static function get_gatsby_content_sync_url_for_post( $post ) {
+		// get the Gatsby Cloud loader url w/ site id
+		$gatsby_content_sync_url = self::get_setting( 'gatsby_content_sync_url' );
+					
+		// create the dynamic path the loader will need
+		$manifest_id = self::get_preview_manifest_id_for_post( $post );
+		$path = "/gatsby-source-wordpress/$manifest_id";
+
+		$url = preg_replace(
```

Comment:
Personal Nit: I need to be better at regular expressions 😆 

I took way to long reading this and making sure I understood it.

Reply:
ha! I'm not great at it either, I always use an online regex building/testing site to construct regex's